unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class_with_nested_generic_class.cpp

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes containing generic
+         classes.
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
```

Comment:
I think this test should probably be amalgamated into `parse_generic_class_with_generic_inner_class` (in the same folder) - though @svorenova may have a more informed opinion?

Reply:
Yes, I agree. Now that I'm looking at it, the java file `GenericClassWithGenericInnerClasses.java`  for that test already has a field with doubly nested generics, so it would be enough to copy the test below to the test file `parse_generic_class_with_generic_inner_classes.cpp
`. I believe this is how the bug was found in the first place (which I forgot about in the meantime ;)).

Reply:
The field in that file is `field6`. Note that actually for this field the bug does not show (the length of the first list of parameters `<..>` is shorter than the name of the class that follows it). It would be good to add a field in the java file that does not parse correctly before the solution, and does parse correctly now.